src/lsm/level_index_iterator.zig

```diff
@@ -1,154 +0,0 @@
-            if (context.direction == .descending) {
-                @panic("TODO Implement descending direction for LevelIndexIterator.");
-            }
-
-            it.* = .{
-                .context = context,
-                .key_exclusive = null,
-                .callback = .none,
-            };
-        }
-
-        /// Calls `callback` with either the next index block or null.
-        /// The block is only valid for the duration of the callback.
-        pub fn next(it: *LevelIndexIterator, callback: Callback) void {
-            assert(it.callback == .none);
-
-            // NOTE We must ensure that between calls to `next`,
-            //      no changes are made to the manifest that are visible to `it.context.snapshot`.
-            const next_table_info = it.context.manifest.next_table(
-                .{
-                    .level = it.context.level,
-                    .snapshot = it.context.snapshot,
-                    .key_min = it.context.key_min,
-                    .key_max = it.context.key_max,
-                    .key_exclusive = it.key_exclusive,
-                    .direction = it.context.direction,
-                },
-            );
-            if (next_table_info) |table_info| {
-                it.key_exclusive = switch (it.context.direction) {
-                    .ascending => table_info.key_max,
-                    .descending => table_info.key_min,
-                };
-                it.callback = .{
-                    .read = .{
-                        .callback = callback,
-                        // Copy table_info so we can hold on to it across `read_block`.
-                        .table_info = table_info.*,
-                    },
-                };
-                it.context.grid.read_block(
-                    .{ .from_local_or_global_storage = on_read },
-                    &it.read,
-                    table_info.address,
-                    table_info.checksum,
-                    .{ .cache_read = true, .cache_write = true },
-                );
-            } else {
-                it.callback = .{ .next_tick = callback };
-                it.context.grid.on_next_tick(on_next_tick, &it.next_tick);
-            }
-        }
-
-        fn on_read(read: *Grid.Read, block: BlockPtrConst) void {
-            const it = @fieldParentPtr(LevelIndexIterator, "read", read);
-            assert(it.callback == .read);
-
-            const callback = it.callback.read.callback;
-            const table_info = it.callback.read.table_info;
-            it.callback = .none;
-
-            callback(it, table_info, block);
-        }
-
-        fn on_next_tick(next_tick: *Grid.NextTick) void {
-            const it = @fieldParentPtr(LevelIndexIterator, "next_tick", next_tick);
-            assert(it.callback == .next_tick);
-
-            const callback = it.callback.next_tick;
-            it.callback = .none;
-            callback(it, null, null);
-        }
-    };
-}
```

Comment:
cc @batiati, this thing here is unused, but I haven't looked whether it _should_ be used